ApiFirst/ApiFirst/Controllers/CostumersController.cs

```diff
@@ -0,0 +1,89 @@
+﻿using DomainModels.Models;
+using DomainServices.Interfaces;
+using Microsoft.AspNetCore.Mvc;
+
+namespace WebApi.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class CustomersController : ControllerBase
+    {
+        private readonly ICustomerRepository _service;
+
+        public CustomersController(ICustomerRepository repo)
+        {
+            _service = repo ?? throw new ArgumentNullException(nameof(repo));
+        }
+
+        [HttpGet("get-by-cpf")]
+        public Customer? GetByCpf([FromQuery] string cpf)
+        {
+            return _service.GetByCpf(cpf);
+        }
+
+
+        [HttpGet("get-all")]
+        public List<Customer> GetAll()
+        {
+            return _service.GetAll();
+        }
+
+        [HttpGet]
+        public Customer? GetById(int id)
+        {
+            return _service.GetById(id);
+        }
+
+        [HttpPost]
+        public IActionResult Create([FromBody] Customer model)
+        {
+            if (_service.Create(model))
+            {
+                return CreatedAtAction(nameof(GetById), new { id = model.Id }, model);
+            }
+            return BadRequest("O Email ou CPF já é usado.");
+        }
+
+        [HttpDelete]
+        public IActionResult Delete(int id)
```

Comment:
É um parâmetro obrigatório então ser enviado via rota e não query
```suggestion
        [HttpDelete("{id}")]
        public IActionResult Delete(int id)
```